middleware/kubernetes/kubernetes.go

```diff
@@ -57,6 +57,10 @@ type AutoPath struct {
 	OnNXDOMAIN     int
 }
 
+type Proxyer interface {
```

Comment:
ah - I was starting to mock out the upstream lookups so those cases could be tested.

I can remove it - its not directly related to this feature.